naga/src/valid/analyzer.rs

```diff
@@ -833,6 +833,26 @@ impl FunctionInfo {
         let mut combined_uniformity = FunctionUniformity::new();
         for statement in statements {
             let uniformity = match *statement {
+                S::Phony(expr) => {
+                    let req = self.expressions[expr.index()].uniformity.requirements;
+                    if self
+                        .flags
+                        .contains(ValidationFlags::CONTROL_FLOW_UNIFORMITY)
+                        && !req.is_empty()
+                    {
+                        if let Some(cause) = disruptor {
+                            return Err(FunctionError::NonUniformControlFlow(req, expr, cause)
+                                .with_span_handle(expr, expression_arena));
+                        }
+                    }
```

Comment:
question: @gfx-rs/naga, does this look right? I'm not sure, since AFAIK the RHS of an assignment shouldn't have any visible effects. Therefore, it shouldn't matter if we have any uniformity obligations here, maybe?